Sources/StreamChatTestTools/Spy.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public protocol Spy: AnyObject {
+    var recordedFunctions: [String] { get set }
+}
```

Comment:
My only concern with this is that we are adding yet another API to how we write tests and mocks. I usually defend using the native XCTest tools as much as we can or use a proper mocking library in this case. It doesn't feel to me this is adding much value in comparison to just adding a property `xxxxCallCount = 0` on a mock and then simply calling `xxxxCallCount += 1` on the desired function. 🤔

Reply:
It is true that it is yet another way of doing it, but it can much better scale than having one xxxxCallCount per each mock. Don't you think?